libarchive/archive_entry_link_resolver.c

```diff
@@ -201,16 +201,26 @@ archive_entry_linkify(struct archive_entry_linkresolver *res,
 		le = find_entry(res, *e);
 		if (le != NULL) {
 			archive_entry_unset_size(*e);
+#if defined(_WIN32) && !defined(__CYGWIN__)
+			archive_entry_copy_hardlink_w(*e,
+			    archive_entry_pathname_w(le->canonical));
+#else
 			archive_entry_copy_hardlink(*e,
 			    archive_entry_pathname(le->canonical));
+#endif
```

Comment:
I debated just doing a deep copy of the `archive_mstring` here, but that required reaching into the "internals" of the `archive_entry` type which felt awkward, especially since I didn't see any examples of this being done elsewhere.